File: EjerciciosFlutter/relacion1/ejercicio8.dart
/*8- Haz un programa en Dart que muestre la tabla de multiplicar de un número entero (entre
1 y 10) que se introducirá por consola. Investiga la función stdin.readLineSync() para
realizar entrada de datos desde consola. ¿Qué ocurre con el sistema “sound type”?.
Soluciónalo utilizando un tipo nullable. Testea la entrada sometiéndola a distintas pruebas
de datos de entrada.*/
import 'dart:io';

void main() {
  bool numValido = false;
  int n=0;
  while (!numValido) {
   int n = devolverEnteroValido("Introduzca un numero entero entre 1 y 10");
    if((n>=1)&&(n<=10)){
      numValido=true;
    }
  }
  print("Tabla de multiplicar del $n");
  for (int i = 1; i <= 10; i++) {
    print("$n*$i=${(i * n)}");
  }
}

int devolverEnteroValido(String mensaje) {
  bool numValido = false;
  int n = 0;
  while (!numValido) {
    stdout.write("${mensaje}\n");
    try {
      n = int.parse(stdin.readLineSync()!);
      return n;
    } catch (ex) {
      print(ex.toString());
    }
  }
}

File: Python/OPTATIVA/Tema 3/Ejercicio8.py
#8. Escribe un programa que recoja un número y calcule si es primo.
num = int(input("Introduzca un numero para saber si es primo:"))
primo = True
i=2

while primo==True and i<num:
    if(num%i==0):
        primo=False
    i+=1
if(num==1):
    primo=False

if(primo == True):
    print("El numero introducido",num,"es primo")
else:
    print("El numero introducido",num,"no es primo")